cmd/infra/powervs/create.go

```diff
@@ -986,33 +986,41 @@ func (infra *Infra) createCloudConnection(options *CreateInfraOptions, client *i
 	return
 }
 
-// setupPowerVSDhcp takes care of setting up dhcp in powervs
-func (infra *Infra) setupPowerVSDhcp(options *CreateInfraOptions, session *ibmpisession.IBMPISession) (err error) {
+// useExistingDHCP returns details of existing DHCP server
+func useExistingDHCP(dhcpServersLen int, dhcpServers models.DHCPServers) (*models.DHCPServerDetail, error) {
+	if dhcpServersLen == 1 {
+		dhcp := dhcpServers[0]
+		return &models.DHCPServerDetail{ID: dhcp.ID, Status: dhcp.Status, Network: dhcp.Network}, nil
```

Comment:
Not comfortable constructing own struct with all the details, it will be good we can run the IBMPIDhcpClient's Get call for the dhcp.ID, this way the content will be same for both create and existing call.

Reply:
can we just return the ID for or the DHCPServer struct and then call a GET call and use that data to be consistent.